lib/msf/core/payload/adapter/fetch.rb

```diff
@@ -210,6 +210,8 @@ def _execute_win(get_file_cmd)
 
   def _execute_nix(get_file_cmd)
     return _generate_fileless(get_file_cmd) if datastore['FETCH_FILELESS']
```

Comment:
As these options seem to be mutually exclusive (`FETCH_FILELESS_PYTHON` won't get executed if `FETCH_FILELESS` is true), could we change the `FETCH_FILELESS` datastore option to an `OptEnum`? That way, we could potentially use:
```
set FETCH_FILELESS none (default?)
set FETCH_FILELESS python3.8+
set FETCH_FILELESS ruby
set FETCH_FILELESS bash
```